.github/workflows/python_tests.yml

```diff
@@ -8,21 +8,20 @@ on:
 
 jobs:
   python-tests:
-
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```

Comment:
I've managed to run the python tests successfully locally using `act --job python-tests`, there is a couple of key things that needed to be added, most important is that i had to add `-DKOMPUTE_OPT_BUILT_IN_VULKAN_HEADER_TAG=v1.2.131` to the `setup.py` file. Is this something that you can run locally as well? 

Testing this across various environments I'm realising something, mainly that it seems there are some issues / incompatibilities that arise from using the `-DKOMPUTE_OPT_BUILT_IN_VULKAN_HEADER_TAG` default value. Before we were assuming the installed vulkan library would be available, but when those files are available it is quite often that the header files are available as well. I think based on this we should be quite confident to use the provided `vulkan.h` headers by default instead of using different in-built ones by default.

It does make sense to allow the header tag to be configurable for the `vulkan.hpp` headers as it seems the compatibility of hpp->h is much more robust than allowing users to specify the version of their `.h` files, as these are quite tightly coupled with the shared library.

It would be great to get thoughts on this, as this would really help with making compatibility simpler "by default", as it would "just work" for the CI as well as local environments.

Reply:
The issue with using the system Vulkan headers is, what if they are more recent than the driver supports?
Exactly this was the reason for me on Fedora 35. Back then I had Vulkan 1.3 headers installed and my mesa-driver supported only 1.2. I also was not able to downgrade my headers since this broke other things on my system.

So, I decided to set the `KOMPUTE_OPT_BUILT_IN_VULKAN_HEADER_TAG` to a sensible version of Vulkan 1.2 and add a version check, which should prevent such a behaviour and notify the user in case something that does not match is detected.

What I can do is add an option like `KOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADERS` and set it to `OFF` by default. There it then would use the system headers by default and the user can change this behaviour in case he needs to.

Reply:
`-DKOMPUTE_OPT_BUILT_IN_VULKAN_HEADER_TAG=v1.2.131` does not work on my end. 
But the python tests work now:
https://github.com/COM8/kompute/actions/runs/2745912792
That's what I have changed: https://github.com/COM8/kompute/commit/b0f6535c4984ddf40e5ff4a7a2b11ad2215c1cab

Reply:
OK interesting that does provide more clarity on the reasoning. I could see the issue both ways - would it be an option to as a default allow it to be inferred from the Vulkan info output with a cap of the highest supported version (I guess 1.2.x)?

Reply:
Hmm ok I see what you mean, ok in that case just using `KOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADERS` would be quite a reasonable approach, let's go for that

Reply:
`KOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADERS` already existed. I fixed a few internal bugs and now it's working like expected. It's set to `OFF` by default. I'm not sure, but this might make some problems when compiling for Android, just so you are aware.